students/meet_shah/lesson09/donor_class.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+from functools import reduce
+
+class Donor:
+
+    def __init__(self, name, donations=None):
+        self._name = name
+        self._donations = [] if donations is None else [donations]
+
+
+    @property
+    def name(self):
+        return self._name
+
+
+    @property
+    def donations(self):
+        return self._donations
+
+
+    def add_donations(self, donation):
+        self._donations.append(donation)
+
+
+    @property
+    def num_donations(self):
+        return len(self._donations)
+
+
+    @property
+    def total_donations(self):
+        #return reduce(lambda a, x: a+x, self._donations, 0)
+        return sum(self._donations)
+
+
+    @property
+    def average_donations(self):
+        return sum(self._donations) * 1.0 / len(self._donations)
+
+
+    @property
+    def information(self):
+        return (self._name, self.total_donations, self.num_donations, self.average_donations)
+
+    def __str__(self):
+        return "{}: {}".format(self._name, self._donations)
+
+
+class DonorDB:
+
+    def __init__(self):
+        self._donors = {}
+
+
+    def update_donor(self, donor):
+        """
+        update_donor update donor's donation, if a donor DNE, this method adds new one to the list
+        """
+        for donation in donor.donations:
+            self._donors.setdefault(donor.name.lower(), Donor(donor.name)).add_donations(donation)
```

Comment:
a donor object can be initialized with a list of donations.

Can you use that to initialize it here without the loop?